### PR TITLE
Support `ceil_mode` attribute for AveragePool, MaxPool ops

### PR DESCRIPTION
**Pull request by robertknight**
For the case of "same" padding there is a quirk that two different but equivalent formulas are given for the output shape in the spec. The main ML frameworks on which ONNX is based (TensorFlow and PyTorch) individually support either "same" padding or a `ceil_mode` argument but not both, so this looks like a corner case that is unimportant.

 - Add `ceil_mode` attribute to MaxPool, AveragePool
 - Implement ceil mode support in `calc_output_size_and_padding`
 - Refactor `calc_output_size_and_padding` tests to use a Default impl to save some repetition

### CODE REVIEW

**Comment by robertknight** [comment]
> There is a note in the spec that says "Sliding windows that would start in the right padded region are ignored". I'm not clear yet what that means

Ah. It relates to an ["Ensure that the last pooling starts inside the image" comment](https://docs.pytorch.org/docs/stable/generated/torch.nn.MaxPool1d.html#:~:text=Ensure%20that%20the%20last%20pooling%20starts%20inside%20the%20image)  in the PyTorch docs.
